.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,39 @@
+# https://github.com/golangci/golangci-lint-action#how-to-use
+name: golangci-lint
+on:
+  push:
+    branches: [main]
+  pull_request:
```

Comment:
FWIW we usually allowlist the base branches of interest.  If the goal here is to pick up the `rhcos-*` branches, wildcards are allowed.